skeleton/generateLookupArray.py

```diff
@@ -1,31 +1,58 @@
-from rotationalOperators import seventhSubIter, eighthSubIter, ninthSubIter, tenthSubIter, eleventhSubIter, twelvethSubIter
+
+from skeleton.rotationalOperators import DIRECTIONLIST
 
 """
-   reference paper
-   http://web.inf.u-szeged.hu/ipcg/publications/papers/PalagyiKuba_GMIP1999.pdf
+Lookuptable is 3Scan's idea of pre-generating a look up array of length (2 ** 26)
+that has all possible configurations of binary strings of length 26 representing
+26 voxels around a voxel at origin in a cube as indices and values at these
+indices saying if the voxel can be removed or not (as it belongs to the boundary
+not the skeleton) as in reference paper
+A Parallel 3D 12-Subiteration Thinning Algorithm Kálmán Palágyi,Graphical Models and Image Processing
+Volume 61, Issue 4, July 1999, Pages 199-221 Attila Kuba, 1999
 """
 
 
-def generateLookuparray(stop, iterationNumber):
+def generateLookupArray(stop, direction):
     """
-       to generate a look up aray the all the 2 ** 26 possible configurations are
-       checked if they satisy one of the 14 templates
-       iterationNumber is the rotation of the cube into one of the directions
-       refer to the paper Parallel 3D thinning algorithm with 12 directions for more details
+    Returns lookuparray
+
+    Parameters
+    ----------
+    stop : int
+    integer describing the length of array
+
+    direction : function
+       function that describes rotation of cube to remove boundary voxels in a different direction
+
+    Returns
+    -------
+    lookuparray : array
+        value at an index of the array = 0 => should not be deleted
+        value at an index of the array = 1 => should be deleted
+
+    Notes
+    ------
+    This program is run once, and the array is saved as lookuparray.npy in
+    the same folder in main function. It doesn't have to be run again unless if templates are changed
+
     """
-    lookuparray = np.zeros(stop, dtype=bool)
-    print(iterationNumber)
+    lookUparray = np.zeros(stop, dtype=bool)
     for item in range(0, stop):
+        # convert the decimal number to a binary string
         neighborValues = [(item >> digit) & 0x01 for digit in range(26)]
+        # voxel at origin/center of the cube should be nonzero, so insert
         neighborValues.insert(13, 1)
+        # 3 x 3 x 3 cube
         neighborMatrix = np.reshape(neighborValues, (3, 3, 3))
+        # if it's a single non zero voxel in the cube
         if np.sum(neighborValues) == 1:
-            lookuparray[item] = 0
+            lookUparray[item] = 0
         else:
-            neighborValues = iterationNumber(neighborMatrix)
+            neighborValues = direction(neighborMatrix)
+            # assign 26 voxels in a 2nd ordered neighborhood of a 3D voxels as 26 alphabet variables
             a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u, v, w, x, y, z = tuple(neighborValues)
```

Comment:
The shouldVoxelBedeleted should have a capital D, and should probably be moved to a subfunction.